marlowe-cli/command/Language/Marlowe/CLI/Command/Format.hs

```diff
@@ -0,0 +1,122 @@
+{-# LANGUAGE BlockArguments #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+
+-----------------------------------------------------------------------------
+--
+-- Module      :  $Headers
+-- License     :  Apache 2.0
+--
+-- Stability   :  Experimental
+-- Portability :  Portable
+--
+-----------------------------------------------------------------------------
+
+-- | Format Marlowe contracts in the CLI tool.
+module Language.Marlowe.CLI.Command.Format (
+  -- * Marlowe CLI Commands
+  FormatCommand (..),
+  runFormatCommand,
+  parseFormatCommand,
+) where
+
+import Control.Monad.Except (MonadError, MonadIO (..))
+import Data.Char (toUpper)
+import Language.Marlowe.CLI.Format (
+  maybeWriteJson,
+  maybeWritePretty,
+  maybeWriteYaml,
+  readContractJson,
+  readContractPretty,
+ )
+import Language.Marlowe.CLI.Types (CliError (..))
+import Options.Applicative qualified as O
+
+-- | Marlowe CLI options for formatting contracts.
+data FormatCommand
+  = -- | Format a contract.
+    Format
+    { inputFile :: Maybe FilePath
+    -- ^ The Marlowe file containing the contract to be formatted.
+    , outputFile :: Maybe FilePath
+    -- ^ The output file for the formatted Marlowe contract.
+    , inFormat :: Maybe Format
+    -- ^ Format of the input Marlowe contract.
+    , outFormat :: Maybe Format
+    -- ^ Format of the output Marlowe contract.
+    }
+  deriving stock (Eq, Show)
+
+-- | Format a Marlowe contract.
+runFormatCommand
+  :: (MonadError CliError m)
+  => (MonadIO m)
+  => FormatCommand
+  -- ^ The command.
+  -> m ()
+  -- ^ Action for runninng the command.
+runFormatCommand Format{..} =
+  do
+    contract <- case inFormat of
+      Just Json -> readContractJson inputFile
+      Just Yaml -> readContractJson inputFile
+      Just Pretty -> readContractPretty inputFile
+      Nothing -> readContractJson inputFile
+    case outFormat of
+      Just Json -> maybeWriteJson outputFile contract
+      Just Yaml -> maybeWriteYaml outputFile contract
+      Just Pretty -> maybeWritePretty outputFile contract
+      Nothing -> maybeWriteJson outputFile contract
+
+-- | Parser for format commands.
+parseFormatCommand :: O.Parser FormatCommand
+parseFormatCommand =
+  Format
+    <$> inFile
+    <*> outFile
+    <*> inFormatParser
+    <*> outFormatParser
+  where
+    inFile =
+      O.optional . O.strOption $
+        mconcat
+          [ O.long "in-file"
+          , O.metavar "MARLOWE_FILE"
+          , O.help "The Marlowe file containing the contract. Stdin if omitted."
+          ]
+    outFile =
+      O.optional . O.strOption $
+        mconcat
+          [ O.long "out-file"
+          , O.metavar "MARLOWE_FILE"
+          , O.help "The Marlowe file containing the output contract. Stdout if omitted."
+          ]
+    inFormatParser =
+      O.optional . O.option formatReader $
+        mconcat
+          [ O.long "in"
```

Comment:
```suggestion
          [ O.long "in-format"
```
Even thought it would be more verbose, it may be good to call this `in-format`, because `in` is a bit ambiguous. Other alternatives would be:
- Having `--to-json`, `--to-yaml`, `--to-marlowe` as a flag instead, have `<in_file>` and `<out_file>` be required, and then infer <in_file> from the content
- Having combined arguments `input.json:json` `output.marlowe:marlowe`

Reply:
Same for `out`

Reply:
Agree, changed to `in-format` and `out-format`. 

I prefer to have those arguments, otherwise also the flags `--from-json`, `--from-yaml` and `--from-marlowe` would be needed in order to read the contract from stdin. That's probably too many flags.

Combined arguments allowing as well `stdin` and `stdout`  instead of the file name would work.

Reply:
Yes, for `stdin` and `stdout` we can either have `-` (dash) mean that (that is pretty standard), or just omit `--in-file` or `--out-file`, but no need to implement that for now